sensor/aosong/dht.go

```diff
@@ -0,0 +1,97 @@
+package aosong
+
+import (
+	"fmt"
+
+	"github.com/ffenix113/zigbee_home/sensor/base"
+	"github.com/ffenix113/zigbee_home/templates/extenders"
+	"github.com/ffenix113/zigbee_home/types/appconfig"
+	dt "github.com/ffenix113/zigbee_home/types/devicetree"
+	"github.com/ffenix113/zigbee_home/types/generator"
+	"github.com/ffenix113/zigbee_home/zcl/cluster"
+	"github.com/ffenix113/zigbee_home/types"
+)
+
+type DHT struct {
+	*base.Base `yaml:",inline"`
+	Pin      types.Pin
+    Variant  string `yaml:"variant"`
+}
+
+func NewDHT() *DHT {
+	return &DHT{
+		Variant: "",
+	}
+}
+
+func (b DHT) String() string {
+	return "Aosong"
+}
+
+func (DHT) Clusters() cluster.Clusters {
+	return []cluster.Cluster{
+		cluster.Temperature{
+			MinMeasuredValue: -40,
+			MaxMeasuredValue: 80,
+			Tolerance:        1,
+		},
+        cluster.NewRelativeHumidity(0, 100),
+	}
+}
+
+func (b DHT) AppConfig() []appconfig.ConfigValue {
+	return []appconfig.ConfigValue{
+		appconfig.NewValue("CONFIG_GPIO").Required(appconfig.Yes),
+		appconfig.CONFIG_DHT.Required(appconfig.Yes),
+	}
+}
+
+func (b DHT) ApplyOverlay(tree *dt.DeviceTree) error {
+
+    pinctrlNode := tree.FindSpecificNode(dt.SearchByLabel(dt.NodeLabelPinctrl))
+	if pinctrlNode == nil {
+		return dt.ErrNodeNotFound(dt.NodeLabelPinctrl)
+	}
+
+    pinLabel := fmt.Sprintf("gpio%d %d (GPIO_PULL_UP | GPIO_ACTIVE_LOW)", b.Pin.Port, b.Pin.Pin)
+
+    //For dht22 or AM2302 devices the string "dht22;" must be added to device tree overlay
+    //See : https://docs.zephyrproject.org/latest/build/dts/api/bindings/sensor/aosong,dht.html
+    //FIXME : Ugly if/else to add the property "dht22;" (from b.Variant) when needed
+    if b.Variant == "" {
+        pinctrlNode.AddNodes(
+            &dt.Node{
+                Name:  "dht22",
+                //FIXME : "_1" hardcoded : can add only one sensor ?
+                Label: "aosong_dht_1",
+                Properties: []dt.Property{
+                    dt.NewProperty("compatible", dt.FromValue("aosong,dht")),
+                    dt.NewProperty("status", dt.FromValue("okay")),
+                    dt.NewProperty("dio-gpios", dt.Angled(dt.Label(fmt.Sprintf(pinLabel)))),
+                },
+            },
+        )
+    } else {
+        pinctrlNode.AddNodes(
+            &dt.Node{
+                Name:  "dht22",
+                //FIXME : "_1" hardcoded : can add only one sensor ?
+                Label: "aosong_dht_1",
+                Properties: []dt.Property{
+                    dt.NewProperty("compatible", dt.FromValue("aosong,dht")),
+                    dt.NewProperty("status", dt.FromValue("okay")),
+                    dt.NewProperty(b.Variant, nil),
+                    dt.NewProperty("dio-gpios", dt.Angled(dt.Label(fmt.Sprintf(pinLabel)))),
+                },
+            },
+        )
+    }
+
+	return nil
+}
+
+func (DHT) Extenders() []generator.Extender {
+	return []generator.Extender{
+		extenders.NewSensor(),
+	}
```

Comment:
Could you please also add GPIO extender, as [SCD4X does](https://github.com/ffenix113/zigbee_home/blob/5267ef5898b91f65a95f8ddfb3e221d1e6d11cf8/sensor/sensirion/scd4x.go#L75-L80)? It will add necessary GPIO import, if it was not added previously.

Reply:
I added a GPIO extender, but (if I understand well your advise) there is no GPIO extender in SCD4X code. I made like on_off sensor code.

Reply:
You are correct, sorry, I misread the extender in SCD4x. Proper link for example is [here](https://github.com/ffenix113/zigbee_home/blob/492a2bc1d2d0ba2354a900a51b3f8e8c68596ed4/sensor/base/ias_zone.go#L50).

You've done it correctly, despite my wrong link, nice work!